### ISSUE TITLE
Build failure: python3Packages.influxdb

### ISSUE THREAD

**Issue by qubitnano** [NixOS/nixpkgs#291415]
### Steps To Reproduce

1. build ``python3Packages.influxdb`` on ``73de017ef2d18a04ac4bfd0c02650007ccb31c2a``

### Build log

```
error: builder for '/nix/store/jnjh0w0lzg3h5wrmycz2jimmi90lqihr-python3.11-influxdb-5.3.1.drv' failed with exit code 1;
       last 10 log lines:
       >
       > influxdb/tests/influxdb08/client_test.py::TestInfluxDBClient::test_write_points_bad_precision
       >   /build/influxdb-5.3.1/influxdb/tests/influxdb08/client_test.py:307: DeprecationWarning: Please use assertRaisesRegex instead.
       >     with self.assertRaisesRegexp(
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED influxdb/tests/dataframe_client_test.py::TestDataFrameClient::test_write_points_from_dataframe_with_tag_escaped - FutureWarning: 'H' is deprecated and will be removed in a future version, p...
       > ===== 1 failed, 221 passed, 2 skipped, 7 deselected, 17 warnings in 1.32s ======
       > /nix/store/3qnm3nwjajgqa771dmi2dnwxrw0kzq5m-stdenv-linux/setup: line 1578: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/jnjh0w0lzg3h5wrmycz2jimmi90lqihr-python3.11-influxdb-5.3.1.drv'.
```

### Additional context

affects at least ceph

```
❯ nix build .#ceph
error: builder for '/nix/store/lbm92fdmqyzys8iri8ddh8fa6qqa92nr-python3.10-influxdb-5.3.1.drv' failed with exit code 1;
       last 10 log lines:
       >
       > influxdb/tests/influxdb08/client_test.py::TestInfluxDBClient::test_write_points_bad_precision
       >   /build/influxdb-5.3.1/influxdb/tests/influxdb08/client_test.py:307: DeprecationWarning: Please use assertRaisesRegex instead.
       >     with self.assertRaisesRegexp(
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED influxdb/tests/dataframe_client_test.py::TestDataFrameClient::test_write_points_from_dataframe_with_tag_escaped - FutureWarning: 'H' is deprecated and will be removed in a future version, p...
       > ===== 1 failed, 221 passed, 2 skipped, 7 deselected, 17 warnings in 1.27s ======
       > /nix/store/3qnm3nwjajgqa771dmi2dnwxrw0kzq5m-stdenv-linux/setup: line 1578: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/lbm92fdmqyzys8iri8ddh8fa6qqa92nr-python3.10-influxdb-5.3.1.drv'.
error: 1 dependencies of derivation '/nix/store/bs73pcfd03r1jcjravs74q4sxf3zbaa1-python3-3.10.13-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rxawkb07jzvafy4921rxyyvdbi2jffr8-ceph-18.2.1.drv' failed to build
```

### Notify maintainers

@fabaff

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.17, NixOS, 24.05 (Uakari), 24.05.20240224.1699da6`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/etc/nix/path/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by SuperSandro2000** [comment]
You don't need to create issues for the sake of creating them. Also please post full logs, the short summary is in most cases useless.

**Comment by SuperSandro2000** [comment]
Already got fixed by #289893
